includes/class-amp-theme-support.php

```diff
@@ -397,12 +398,54 @@ public static function finish_init() {
 				self::redirect_non_amp_url( current_user_can( 'manage_options' ) ? 302 : 301 );
 			}
 
+			if ( MobileRedirectManager::is_enabled() && is_amp_available() ) {
```

Comment:
Maybe more of the logic in this block should move to method(s) in `MobileRedirectManager`?

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4796/commits/5cb756ed4759dac86d1b77441d3e0e741138cac2.